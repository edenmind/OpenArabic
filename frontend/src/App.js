import { BrowserRouter, Route, Routes } from 'react-router-dom'

import About from './screens/About'
import { AuthorAdd } from './screens/Authors/AuthorAdd'
import { AuthorUpdate } from './screens/Authors/AuthorUpdate'
import { Authors } from './screens/Authors/Authors'
import { Categories } from './screens/Categories/Categories'
import { CategoryAdd } from './screens/Categories/CategoryAdd'
import { CategoryUpdate } from './screens/Categories/CategoryUpdate'
import Contact from './screens/Contact'
import Home from './screens/Home'
import Privacy from './screens/Privacy'
import Text from './screens/Texts/Text'
import Texts from './screens/Texts/Texts'

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path='/' element={<Home />}>
          <Route index element={<Home />} />
        </Route>
        <Route path='/contact' element={<Contact />} />
        <Route path='/about' element={<About />} />
        <Route path='/texts' element={<Texts />} />
        <Route path='/texts/:id' element={<Text />} />
        <Route path='/authors' element={<Authors />} />
        <Route path='/authors/add' element={<AuthorAdd />} />
        <Route path='/authors/update/:id' element={<AuthorUpdate />} />
        <Route path='/categories/add' element={<CategoryAdd />} />
        <Route path='/categories/update/:id' element={<CategoryUpdate />} />
        <Route path='/categories' element={<Categories />} />
        <Route path='/privacy' element={<Privacy />} />
      </Routes>
    </BrowserRouter>
  )
}
