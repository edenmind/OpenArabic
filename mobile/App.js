import React from 'react'
import PAPERTHEME from './constants/paper-theme.js'
import { Provider as PaperProvider } from 'react-native-paper'
import { Provider } from 'react-redux'
import Root from './routes/root.js'
import { store } from './redux/store.js'
import * as Sentry from 'sentry-expo'
import ErrorBoundary from 'react-native-error-boundary'

export default function App() {
  Sentry.init({
    dsn: 'https://5c02aed50b5d40d198688b1aa145cf29@o1414696.ingest.sentry.io/6755037',
    enableInExpoDevelopment: true
  })

  return (
    <ErrorBoundary>
      <Provider store={store}>
        <PaperProvider theme={PAPERTHEME}>
          <Root />
        </PaperProvider>
      </Provider>
    </ErrorBoundary>
  )
}
