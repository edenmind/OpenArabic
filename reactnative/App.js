import "react-native-gesture-handler";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { StyleSheet } from "react-native";
import { Provider as PaperProvider } from "react-native-paper";
import { createMaterialBottomTabNavigator } from "@react-navigation/material-bottom-tabs";
import TextNavigator from "./screens/Texts/TextNavigator";
import { SettingsScreen } from "./screens/Settings/Settings";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";

const Tab = createMaterialBottomTabNavigator();

export default function App() {
  return (
    <PaperProvider>
      <NavigationContainer>
        <Tab.Navigator>
          <Tab.Screen
            name="Text"
            component={TextNavigator}
            options={{
              tabBarLabel: "Texts",
              tabBarIcon: ({ color }) => (
                <MaterialCommunityIcons name="text" color={color} size={26} />
              ),
            }}
          />
          <Tab.Screen
            name="Settings"
            component={SettingsScreen}
            options={{
              tabBarLabel: "Settings",
              tabBarIcon: ({ color }) => (
                <MaterialCommunityIcons
                  name="cog-outline"
                  color={color}
                  size={26}
                />
              ),
            }}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </PaperProvider>
  );
}

export const style = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
