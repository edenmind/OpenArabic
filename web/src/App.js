import { BrowserRouter, Route, Routes } from 'react-router-dom'

import About from './screens/About'
import { AuthorAdd } from './screens/AuthorAdd'
import { AuthorUpdate } from './screens/AuthorUpdate'
import { Authors } from './screens/Author'
import { Categories } from './screens/Categories'
import Category from './screens/Category'
import { CategoryAdd } from './screens/CategoryAdd'
import { CategoryUpdate } from './screens/CategoryUpdate'
import Home from './screens/Home'
import Privacy from './screens/Privacy'
import SingleText from './screens/SingleText'
import Texts from './screens/Texts'
import TextsAdd from './screens/TextAdd'

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path='/' element={<Home />} />
        <Route path='/about' element={<About />} />
        <Route path='/texts/' element={<Texts />} />
        <Route path='/texts/add' element={<TextsAdd />} />
        <Route path='/texts/categories/:id' element={<Categories />} />
        <Route path='/texts/:id' element={<SingleText />} />
        <Route path='/authors' element={<Authors />} />
        <Route path='/authors/add' element={<AuthorAdd />} />
        <Route path='/authors/update/:id' element={<AuthorUpdate />} />
        <Route path='/categories/add' element={<CategoryAdd />} />
        <Route path='/categories/update/:id' element={<CategoryUpdate />} />
        <Route path='/categories' element={<Category />} />
        <Route path='/privacy' element={<Privacy />} />
      </Routes>
    </BrowserRouter>
  )
}
