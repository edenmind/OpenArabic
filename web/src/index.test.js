import React from 'react'
import ReactDOM from 'react-dom'
import App from './app.js'

jest.mock('react-dom', () => ({ render: jest.fn() }))

describe('Application root', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    global.document.getElementById = (id) => id === 'root' && div
    expect(ReactDOM.render).toHaveBeenCalled()
  })
})
