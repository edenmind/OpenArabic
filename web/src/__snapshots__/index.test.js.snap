// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application root should match snapshot 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      <App />,
      <div />,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
